### ISSUE TITLE
inconsistent pose and depth result

### ISSUE THREAD

**Issue by LinRui9531** [ClementPinard/SfmLearner-Pytorch#40]
I trained model with this commend:
python train.py /path/to/the/formatted/data/ -b4 -m0 -s2.0 --epoch-size 1000 --sequence-length 5 
but my result was not as good as yours.

Abs   rel | Sq   rel | RMSE    | RMSE   log | A1        | A2       | A3        |
 0.3794   | 3.8478 | 10.6573 | 0.5115        | 0.3702 | 0.6512 |   0.8179 | 

pose 
             ate          re
seq 09  0.0308 1.2759
seq 10  0.0261  1.3253

Can you tell me the possible reason? or did you train with gt?

**Comment by ClementPinard** [comment]
Hello,

The gt is only here for Validation, and helps early stopping

If you don't have gt on validation set, I advise you to record a network every 10 epochs or so and then try them all to see which is the best.

The thing is this training workflow is not very robust to overfitting, and quality worsens at some point, the key is to know when it happens.

So instead of testing all network versions on test, I decided to make a vlidation set with gt on which we can choose the best network, so that we don't officially optimize on the test set.
